packages/flutter/lib/src/cupertino/app.dart

```diff
@@ -680,95 +682,84 @@ class _CupertinoAppState extends State<CupertinoApp> {
   }
 }
 
-enum _CupertinoInspectorButtonVariant { filled, toggle, iconOnly }
-
-class _CupertinoInspectorButton extends StatelessWidget {
-  const _CupertinoInspectorButton({
-    required this.onPressed,
-    required this.semanticLabel,
-    required this.icon,
-    this.buttonKey,
-  }) : _variant = _CupertinoInspectorButtonVariant.filled,
-       _toggledOn = null;
+class _CupertinoInspectorButton extends InspectorButton {
+  const _CupertinoInspectorButton.filled({
+    required super.onPressed,
+    required super.semanticLabel,
+    required super.icon,
+    super.buttonKey,
+  }) : super.filled();
 
   const _CupertinoInspectorButton.toggle({
-    required this.onPressed,
-    required this.semanticLabel,
-    required this.icon,
-    bool toggledOn = true,
-  }) : buttonKey = null,
-       _variant = _CupertinoInspectorButtonVariant.toggle,
-       _toggledOn = toggledOn;
+    required super.onPressed,
+    required super.semanticLabel,
+    required super.icon,
+    super.toggledOn,
+  }) : super.toggle();
 
   const _CupertinoInspectorButton.iconOnly({
-    required this.onPressed,
-    required this.semanticLabel,
-    required this.icon,
-  }) : buttonKey = null,
-       _variant = _CupertinoInspectorButtonVariant.iconOnly,
-       _toggledOn = null;
-
-  final VoidCallback onPressed;
-  final String semanticLabel;
-  final IconData icon;
-  final GlobalKey? buttonKey;
-  final _CupertinoInspectorButtonVariant _variant;
-  final bool? _toggledOn;
-
-  static const double _buttonSize = 32.0;
-  static const double _buttonIconSize = 18.0;
+    required super.onPressed,
+    required super.semanticLabel,
+    required super.icon,
+  }) : super.iconOnly();
 
   @override
   Widget build(BuildContext context) {
     final Icon buttonIcon = Icon(
       icon,
       semanticLabel: semanticLabel,
-      size: _variant == _CupertinoInspectorButtonVariant.iconOnly ? _buttonSize : _buttonIconSize,
-      color: _foregroundColor(context),
+      size:
```

Comment:
use the `InspectorButton.currentIconSize` getter instead

Reply:
oops done!